packages/runtime-dom/src/directives/vOn.ts

```diff
@@ -10,6 +10,19 @@ import { hyphenate, isArray } from '@vue/shared'
 const systemModifiers = ['ctrl', 'shift', 'alt', 'meta']
 
 type KeyedEvent = KeyboardEvent | MouseEvent | TouchEvent
+type ModifierGuardsKeys =
+  | 'stop'
+  | 'prevent'
+  | 'self'
+  | 'ctrl'
+  | 'shift'
+  | 'alt'
+  | 'meta'
+  | 'left'
+  | 'middle'
+  | 'right'
+  | 'exact'
+  | ''
 
 const modifierGuards: Record<
   string,
```

Comment:
This should use `ModifierGuardsKeys` as well

Reply:
Done.